ops/model.py

```diff
@@ -43,32 +59,51 @@ def __init__(self, unit_name, meta, backend):
         self.storages = StorageMapping(list(meta.storages), self._backend)
         self._bindings = BindingMapping(self._backend)
 
-    def get_unit(self, unit_name):
+    def get_unit(self, unit_name: str) -> 'Unit':
+        """Get an arbitrary unit by name.
+
+        Internally this uses a cache, so asking for the same unit two times will
+        return the same object.
+        """
         return self._cache.get(Unit, unit_name)
 
-    def get_app(self, app_name):
+    def get_app(self, app_name: str) -> 'Application':
+        """Get an application by name.
+
+        Internally this uses a cache, so asking for the same application two times will
+        return the same object.
+        """
         return self._cache.get(Application, app_name)
 
-    def get_relation(self, relation_name, relation_id=None):
+    def get_relation(
+            self, relation_name: str,
+            relation_id: typing.Optional[int] = None) -> 'Relation':
         """Get a specific Relation instance.
 
-        If relation_id is given, this will return that Relation instance.
-
         If relation_id is not given, this will return the Relation instance if the
         relation is established only once or None if it is not established. If this
         same relation is established multiple times the error TooManyRelatedAppsError is raised.
+
+        Parameters:
+            relation_name: The name of the endpoint for this charm
+            relation_id: An identifier for a specific relation. Used to disambiguate when a
+                given application has more than one relation on a given endpoint.
+        Raises:
+            TooManyRelatedAppsError: is raised if there is more than one relation to the
+                supplied relation_name and no relation_id was supplied
         """
         return self.relations._get_unique(relation_name, relation_id)
 
-    def get_binding(self, binding_key):
+    def get_binding(self, binding_key: typing.Union[str, 'Relation']) -> 'Binding':
         """Get a network space binding.
 
-        binding_key -- The relation name or instance to obtain bindings for.
-
-        If binding_key is a relation name, the method returns the default binding for that
-        relation. If a relation instance is provided, the method first looks up a more specific
-        binding for that specific relation ID, and if none is found falls back to the default
-        binding for the relation name.
+        Parameters:
+            binding_key -- The relation name or instance to obtain bindings for.
```

Comment:
```suggestion
            binding_key: The relation name or instance to obtain bindings for.
```
there are also a few leftover ` -- `s in `model.py` from before we agreed on the doc format, which I think would be opportune to fix while you're here.

Reply:
thanks for the catch. A couple of them were on classes that I was considering private, but its still good to clean them up.
Since you pointed this one out, I also made the mistake of calling it "Parameters", it is supposed to be "Args".